lib/bazel.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Bazel support for protoc_plugin.
+library protoc_bazel;
+
+import 'package:path/path.dart' as p;
+import 'protoc.dart' show SingleOptionParser, DefaultOutputConfiguration;
+
+/// Dart protoc plugin option for Bazel packages.
+///
+/// This option takes a semicolon-separated list of Bazel package name to repo
+/// path pairs in `name|path` form.
+const bazelOptionId = 'BazelPackages';
+
+/// Summary information about a Bazel package.
+class BazelPackage {
+  final String name;
+  final String path;
+
+  BazelPackage(this.name, String path) : path = p.normalize(path);
+}
+
+/// Parser for the `BazelPackages` option.
+class BazelOptionParser implements SingleOptionParser {
+  final Map<String, BazelPackage> packages;
+
+  BazelOptionParser(this.packages);
+
+  @override
+  void parse(String name, String value, onError(String message)) {
+    if (value == null) {
+      onError('Invalid $bazelOptionId option. Expected a non-empty value.');
+      return;
+    }
+
+    for (var entry in value.split(';')) {
+      var pair = entry.split('|');
+      if (pair.length != 2) {
```

Comment:
Alternatively, we could split in three here: `package_name|root_path|lib_path`, where `root_path` is the path relative to which we locate input `.proto` files and `lib_path` is the path relative to which we emit the generated `.pb.dart` outputs.

That has the advantage that it avoids encoding any Bazel-specific path logic. Admittedly, there aren't many alternative behaviours I could imagine.